### PR TITLE
Feature/first

### PR DESCRIPTION
**Pull request by bakspace-itk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,21 @@
...
-When the robot is run from OpenOrchestrator the `main.py` file is run which results
-in the following:
-1. The working directory is changed to where `main.py` is located.
-2. A virtual environment is automatically setup with the required packages.
-3. The framework is called passing on all arguments needed by [OpenOrchestrator](https://github.com/itk-dev-rpa/OpenOrchestrator).
+1. Setup a trigger and Miralix Shared Key in OpenOrchestrator.
+3. Setup a case in GetOrganized, making sure metadata is setup properly.
+2. Set Miralix queues and case number as parameters to the trigger: 
+```
+{"case_number": "EMN-2024-123456", "target_queues":["89403330 Opkrævningen P-Gap","89403330 Opkrævningen P-Gap Boliglån tast 2")]}
```
Comment (ghbm-itk) [inline README.md]: Are those real phone numbers and case numbers?

File: robot_framework/config.py [hunk 2]
```diff
@@ -14,14 +14,20 @@
 # Constant/Credential names
 ERROR_EMAIL = "Error Email"
 
+# Miralix
+SSK = "Miralix Shared Key"
```
Comment (ghbm-itk) [inline robot_framework/config.py]: SSK?

File: robot_framework/process.py [hunk 3]
```diff
@@ -1,8 +1,59 @@
...
 
 
 def process(orchestrator_connection: OrchestratorConnection) -> None:
     """Do the primary process of the robot."""
     orchestrator_connection.log_trace("Running process.")
+
+    get_organized_login = orchestrator_connection.get_credential(config.GO_CREDENTIALS)
+    session = get_organized_api.create_session(config.GO_API, get_organized_login.username, get_organized_login.password)
+    miralix_password = orchestrator_connection.get_credential(config.SSK).password
+    case_id = json.loads(orchestrator_connection.process_arguments)["case_number"]
```
Comment (ghbm-itk) [inline robot_framework/process.py]: Is it a number or an id? ;)

File: robot_framework/process.py [hunk 4]
```diff
@@ -1,8 +1,59 @@
...
+
+    #  Check queue elements for highest call ID previously downloaded
+    queue_elements = orchestrator_connection.get_queue_elements(config.QUEUE_NAME, status=QueueStatus.DONE)
+    highest_id = last_download = 0
```
Comment (ghbm-itk) [inline robot_framework/process.py]:
```suggestion
    max(int(queue_element.reference) for queue_element in queue_elements)
```

File: robot_framework/process.py [hunk 5]
```diff
@@ -1,8 +1,59 @@
...
+    for queue_element in queue_elements:
+        last_download = max(highest_id, int(queue_element.reference))
```
Comment (ghbm-itk) [inline robot_framework/process.py]: highest_id is always 0, right?

File: robot_framework/process.py [hunk 6]
```diff
@@ -1,8 +1,59 @@
...
+
+    #  Get list of recordings that have a higher ID than the previous highest, and sort them
+    recordings = miralix_api.recordings_for_process(orchestrator_connection, last_download)
+    recordings.sort(key=lambda recording: recording["QueueCallId"])
+
+    #  Run through each recording, download file data and send to GetOrganized
+    for i, recording in enumerate(recordings):
+        call_id = recording["QueueCallId"]
+        filename = miralix_api.get_filename(recording)
+        print(f"{i+1}/{len(recordings)} - Call ID {call_id} being saved as {filename}")
```
Comment (ghbm-itk) [inline robot_framework/process.py]: Should this be a log?

File: robot_framework/process.py [hunk 7]
```diff
@@ -1,8 +1,59 @@
...
+                                          agent_name=recording["AgentName"],
+                                          date_string=recording["ConversationStartedUtc"])
+
+        orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.DONE)
+
+
+if __name__ == '__main__':
+    conn_string = os.getenv("OpenOrchestratorConnString")
+    crypto_key = os.getenv("OpenOrchestratorKey")
+    oc = OrchestratorConnection("Miralix Nedhentning", conn_string, crypto_key, '{"case_number": "EMN-2024-033020", "target_queues":["89403330 Opkrævningen P-Gap","89403330 Opkrævningen P-Gap Boliglån tast 2", "89404130 BS - Kørekort P-GAP", "89402000 Aarhus Kommunes Hovednummer NPS", "89402088 Janni testnummer NPS", "89402260 BS - Vielseskontoret NPS"]}')
```
Comment (ghbm-itk) [inline robot_framework/process.py:56]: Is this data we want in git?

File: robot_framework/get_organized/get_organized_api.py [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        username: Username for login.
+        password: Password for login.
+
+    Returns:
+        Return the session object
+    """
+    session = Session()
+    session.headers.setdefault("Content-Type", "application/json")
+    session.auth = HttpNtlmAuth(username, password)
+    session.post(apiurl, timeout=config.GO_TIMEOUT)
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py]: What is the point of this post request?

File: robot_framework/get_organized/get_organized_api.py [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+        file: Bytearray of file to upload.
+        case: Case name already present in GO.
+        filename: Name of file when saved in GO.
+        agent_name: Agent name, used for creating a folder in GO. Defaults to None.
+        date_string: A date to add as metadata to GetOrganized. Defaults to None.
+
+    Returns:
+        Return response text and session token.
+    """
+    url = apiurl + "/_goapi/Documents/AddToCase"
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py:43]: url join?

File: robot_framework/get_organized/get_organized_api.py [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+        "Bytes": file,
+        "CaseId": case,
+        "SiteUrl": f"{apiurl}/cases/EMN/{case}",
+        "ListName": "Dokumenter",
+        "FolderPath": agent_name,
+        "FileName": filename,
+        "Metadata": f"<z:row xmlns:z='#RowsetSchema' ows_Dato='{date_string}'/>",
+        "Overwrite": True
+    }
+    response = session.post(url, data=json.dumps(payload), timeout=config.GO_TIMEOUT)
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py:54]: Should we raise an HTTP error here?

Comment (ghbm-itk) [reply robot_framework/get_organized/get_organized_api.py:54]: Same question for all requests

File: robot_framework/get_organized/get_organized_api.py [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+    Returns:
+        Return the response and session objects.
+    """
+    url = apiurl + "/_goapi/Cases/"
+    payload = {
+        'CaseTypePrefix': 'EMN',
+        'MetadataXml': f'<z:row xmlns:z="#RowsetSchema" ows_Title="{title}" ows_CaseStatus="Åben"/>',
+        'ReturnWhenCaseFullyCreated': False
+        }
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py]: :(

File: robot_framework/get_organized/get_organized_api.py [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+    # Send the POST request
+    response = session.post(url, data=json.dumps(payload), timeout=config.GO_TIMEOUT)
+    # Return the response
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py]: This comment smells of AI 😅

File: robot_framework/get_organized/get_organized_api.py [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+    Returns:
+        Return the response and session objects.
+    """
+    url = apiurl + "/_goapi/Cases/CloseCase"
+    payload = {"CaseId": case_number}
+    response = session.post(url, data=payload, timeout=config.GO_TIMEOUT)
+    return response.text, session
+
+
+def journalize_document(apiurl, doc_id, session) -> tuple[str, Session]:
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py]: Is it journalized or finalized?

File: robot_framework/get_organized/get_organized_api.py [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+        "OperationName": "OperationNameFromGoApi",
+        "Message": message,
+        "LogLevel": "1"
+    }
+    response = session.post(url, json.dumps(data))
+    return response, session
+
+
+if __name__ == "__main__":
+    print("main")
```
Comment (ghbm-itk) [inline robot_framework/get_organized/get_organized_api.py]: 🥳

**Review by ghbm-itk — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (bakspace-itk) [reply README.md]: Real phone numbers, fake case number. Fixed.

File: robot_framework/get_organized/get_organized_api.py (hunk 8 above)
Comment (ghbm-itk) [reply robot_framework/get_organized/get_organized_api.py]: ?

**Review by ghbm-itk — APPROVED** [review]
There are still some unresolved comments but otherwise it looks good

**Pull request merged**